lib/src/input_validation/validate_function.dart

```diff
@@ -1,19 +1,27 @@
 import 'package:advanced_calculation/src/input_validation/error_state.dart';
 import 'package:advanced_calculation/src/input_validation/pattern.dart';
+import 'package:advanced_calculation/src/parse/expression_parser.dart';
+import 'package:advanced_calculation/src/translator/translate_pattern.dart';
 
 import 'start_state.dart';
 import 'state.dart';
 
 class ValidateFunction {
+  ExpressionParser parser = ExpressionParser();
+
   static final List<String> validFunctions = ["ln","log","sin","cos","tan", "abs", "csc","sec", "cot", "sqrt", "sinh", "cosh", "tanh",
     "asin", "acos", "atan", "acsc", "asec", "acot", "csch", "sech", "coth", "ceil","asinh", "acosh", "atanh", "acsch", "asech",
-    "acoth", "floor", "round", "trunc", "fract"];
+    "acoth", "floor", "round", "trunc", "fract", "√"];
   static final List<String> multiParamFunctions = ["max", "min", "gcd", "lcm"];
-
+  static final List<String> operands = ['*','/','-','+','(',')','^',',','²'];
 
   List<String> _sanitizeInput(String input){
-    List<String> sanitizedInput = input.split(" ").where((item) => item.isNotEmpty).toList();
-    sanitizedInput.add('=');
+    String trimmed = input;
+    if(input.endsWith(',')){
+      trimmed = input.substring(input.length - 1); // remove the negative
+    }
```

Comment:
adding '=' above already takes care of this case, can remove this if statement lines

Reply:
the input wont have = in it as its directly form the user

Reply:
Then remove the if statements from the state pattern where the check happens for '='

Reply:
will do